File: javascript-oop/mixin-testing/models/student.js
class Student {
    constructor(firstname, lastname, homework) {
        this.firstname = firstname;
        this.lastname = lastname;
        this.homework = homework;
    }

    get firstname() {
        return this._firstname;
    }

    set firstname(firstname) {
        this._firstname = firstname;
    }

    get lastname() {
        return this._lastname;
    }

    set lastname(lastname) {
        this._lastname = lastname;
    }

    get homework() {
        return this._homework;
    }

    set homework(homework) {
        this._homework = homework;
    }
}

module.exports = Student;

File: DataBases/EntityFrameworkCodeFirst/EntityFrameworkCodeFirst/EntityFrameworkCodeFirst.Models/StudentSystem/Student.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EntityFrameworkCodeFirst.Models.StudentSystem
{
    public class Student
    {
        private ICollection<Course> courses;
        private ICollection<Homework> homeoworks;


        public Student()
        {
            this.courses = new HashSet<Course>();
            this.homeoworks = new HashSet<Homework>();
        }

        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(20, MinimumLength = 3)]
        public string FirstName { get; set; }

        [Required]
        [StringLength(20, MinimumLength = 3)]
        public string LastName { get; set; }

        [Index(IsClustered = false, IsUnique = true)]
        [StringLength(10, MinimumLength = 10)]
        public string StudentNumber { get; set; }

        public virtual ICollection<Course> Courses
        {
            get { return this.courses; }
            set { this.courses = value; }
        }

        public virtual ICollection<Homework> Homeworks
        {
            get { return this.homeoworks; }
            set { this.homeoworks = value; }
        }
    }
}
